x-pack/legacy/plugins/siem/public/containers/authentications/authentications_over_time/index.tsx

```diff
@@ -73,7 +77,7 @@ class AuthenticationsOverTimeComponentQuery extends QueryTemplate<
             from: startDate!,
             to: endDate!,
           },
-          defaultIndex: chrome.getUiSettingsClient().get(DEFAULT_INDEX_KEY),
+          defaultIndex: kibana.services.uiSettings.get<string[]>(DEFAULT_INDEX_KEY),
```

Comment:
👍 Nice to see the new API. Like it takes a `<string[]>` as a template.

Reply:
What do you think about exporting "bound" generic types internally? 

```ts
const indexPatterns = useUiSetting<string[]>(DEFAULT_INDEX_KEY);
// vs
const indexPatterns = useIndexPatterns();
```
In both cases `indexPatterns` would be typed as `string[]`. I already did that for our own [useKibana](https://github.com/elastic/kibana/pull/52783/files#diff-cc9c9c4c132f135055123b983c661aa2R23) hook to reduce boilerplate, and it makes sense for these common settings that we retrieve all over the place, too (DEFAULT_INDEX_KEY, DEFAULT_KBN_VERSION)

Reply:
I think the bound version reads much better

Reply:
NB this new typed API has so far surfaced [one type error](https://github.com/elastic/kibana/pull/52783/files#diff-2131f68a5c9a219e79bfeee91f992c27R16) that I was excited might be a bug, but the client/endpoint that ultimately use it seem to handle the array interpolation just fine? 🤷‍♂ 

Also, we don't seem to have consistent jargon for this concept of an array of index patterns: it's mainly called `defaultIndex`(not plural?), but I've also seen `defaultIndexPattern`, `indicesConfig`, etc. Is the lack of pluralization due to the fuzziness of the elastic API? Is an array of index patterns also an index pattern?

Reply:
I think it's influenced by the ES API which in most cases takes a variable called index which is an array.

Reply:
FYI: It looks like this broad context might be [replaced by more declarative hooks](https://github.com/elastic/kibana/issues/53029#issuecomment-566776255) in the near future, so wrapping our current usage is definitely the way to go given our simple use case.